// Course: CS6381 Distributed Systems Principles
// Instructor: Aniruddha Gokhale
// Created: Spring 2023
//
// Purpose:
// Describe an initial schema for serializing topic names and their values. For assignment 1,
// we are just using strings. But later assignments we will enhance it.

// Let us use the Version 3 syntax
syntax = "proto3";

// Format for a message sent by the publisher of a certain topic
message topicMessage
{
    string topic = 1;
    string content = 2;
    string timestamp = 3;
}